ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/commons/parameters/UnoptimizedCnecParameters.java

```
@@ -8,36 +8,38 @@
 import com.farao_community.farao.data.crac_api.cnec.FlowCnec;
+import com.farao_community.farao.data.crac_api.range_action.PstRangeAction;
 import com.farao_community.farao.rao_api.parameters.RaoParameters;
 import com.farao_community.farao.search_tree_rao.castor.parameters.SearchTreeRaoParameters;
-import com.farao_community.farao.search_tree_rao.commons.RaoUtil;
 
+import java.util.Map;
 import java.util.Set;
 
 /**
  * @author Joris Mancini {@literal <joris.mancini at rte-france.com>}
+ * @author Godelaine de Montmorillon {@literal <godelaine.demontmorillon at rte-france.com>}
  */
 public class UnoptimizedCnecParameters {
     private final Set<String> operatorNotToOptimize;
-    private final double highestThresholdValue;
+    private final Map<FlowCnec, PstRangeAction> unoptimizedCnecsInSeriesWithPsts;
 
-    public UnoptimizedCnecParameters(Set<String> operatorNotToOptimize, double highestThresholdValue) {
+    public UnoptimizedCnecParameters(Set<String> operatorNotToOptimize, Map<FlowCnec, PstRangeAction> unoptimizedCnecsInSeriesWithPsts) {
         this.operatorNotToOptimize = operatorNotToOptimize;
-        this.highestThresholdValue = highestThresholdValue;
+        this.unoptimizedCnecsInSeriesWithPsts = unoptimizedCnecsInSeriesWithPsts;
     }
 
-    public Set<String> getOperatorsNotToOptimize() {
-        return operatorNotToOptimize;
+    public Map<FlowCnec, PstRangeAction>  getUnoptimizedCnecsInSeriesWithPsts() {
+        return unoptimizedCnecsInSeriesWithPsts;
     }
 
-    public double getHighestThresholdValue() {
-        return highestThresholdValue;
+    public Set<String> getOperatorsNotToOptimize() {
+        return operatorNotToOptimize;
     }
 
-    public static UnoptimizedCnecParameters build(RaoParameters raoParameters, Set<String> operatorsNotSharingCras, Set<FlowCnec> flowCnecs) {
-
+    // unoptimizedCnecsInSeriesWithPsts and operatorNotToOptimize cannot be activated together.
```

you should throw an exception if both are activated